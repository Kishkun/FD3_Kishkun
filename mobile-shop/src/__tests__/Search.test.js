import {mount, shallow} from "enzyme";
import React from "react";
import ConnectedSearch, {Search} from "../components/search/Search";
import {Provider} from "react-redux";
import configureMockStore from "redux-mock-store";

test("testing search component", () => {

    let phones = [
        {
            "id": "1",
            "categoryId": "1",
            "name": "Apple XR",
            "description":
                "The Apple iPhone XR is a smartphone that was tested with the iOS 12.0. 1 operating system. This model weighs 6.9 ounces, has a 6.1 inch touch screen display, 12-megapixel main camera, and 7-megapixel selfie camera. It comes with 3GB of RAM.",
            "price": 823,
            "image": "/store/uploads/apple-iphone-xr-new.jpg",
            "cpu": "A12 Bionic chip",
            "camera": "12MP Wide camera",
            "size": "150.9 x 75.7 x 8.3 mm (5.94 x 2.98 x 0.33 in)",
            "weight": "194 g (6.84 oz)",
            "display": "6.1-inch",
            "battery": "Non-removable Li-Ion 2942 mAh battery (11.16 Wh)",
            "memory": "64GB, 128GB and RAM 4 GB"
        }
    ];

    const initialState = {
        ids: phones
    };
    const mockStore = configureMockStore();
    let store, wrapper;

    store = mockStore(initialState);
    wrapper = mount(<Provider store={store}><ConnectedSearch/></Provider>);

    expect(wrapper).toMatchSnapshot();

    const searchEl = wrapper.find("#search");
    searchEl.instance().value = "apple";
    const searchSubmit = wrapper.find("#search-form");
});