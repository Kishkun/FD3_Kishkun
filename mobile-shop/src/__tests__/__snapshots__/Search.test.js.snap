// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing search component 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(Search)>
    <Search
      searchPhone={[Function]}
    >
      <div
        className="well blosd"
      >
        <h3
          className="lead"
        >
          Quick shop
        </h3>
        <div
          className="input-group"
        >
          <form
            id="search-form"
            onSubmit={[Function]}
          >
            <input
              className="form-control"
              defaultValue=""
              id="search"
              onChange={[Function]}
              placeholder="Search..."
              type="text"
            />
          </form>
          <span
            className="input-group-btn"
          >
            <button
              className="btn btn-default"
            >
              <i
                className="fa fa-search"
              />
            </button>
          </span>
        </div>
      </div>
    </Search>
  </Connect(Search)>
</Provider>
`;
